equistore-core/src/io/npy_header.rs

```diff
@@ -496,7 +632,34 @@ impl From<WriteHeaderError> for crate::Error {
     fn from(error: WriteHeaderError) -> Self {
         match error {
             WriteHeaderError::Io(e) => crate::Error::Io(e),
-            WriteHeaderError::Format(e) => crate::Error::Serialization(e.to_string()),
+            WriteHeaderError::Format(e) => crate::Error::Serialization(e),
         }
     }
 }
+
+#[cfg(test)]
```

Comment:
Shouldn't there be also a test for the writing?

Reply:
This is the `to_dict_literal()` checks below. I could rename the test function name

Reply:
No its fine.